triggers/service/src/main/scala/com/digitalasset/daml/lf/engine/trigger/ServiceMain.scala

```diff
@@ -295,8 +294,33 @@ object ServiceMain {
           )
         val system: ActorSystem[Server.Message] =
           ActorSystem(Server("localhost", 8080, ledgerConfig, dar), "TriggerService")
-        StdIn.readLine()
-        system ! Server.Stop
+        // timeout chosen at random, change freely if you see issues
+        implicit val timeout: Timeout = 15.seconds
+        implicit val scheduler: Scheduler = system.scheduler
+        implicit val ec: ExecutionContext = system.executionContext
+
+        // Shutdown gracefully on SIGINT.
+        val serviceF: Future[ServerBinding] =
+          system.ask((ref: ActorRef[ServerBinding]) => Server.GetServerBinding(ref))
+        Statement.discard {
+          sys.addShutdownHook {
+            system ! Server.Stop
+            serviceF.onComplete {
+              case Success(_) =>
+                system.log.info("Server is offline, the system will now terminate")
+              case Failure(ex) =>
+                system.log.info("Failure encountered shutting down the server: " + ex.toString)
+            }
+            Statement.discard { Await.ready(serviceF, 5.seconds) }
+          }
+        }
     }
   }
 }
+
+object Statement {
```

Comment:
I find this more confusing than helpful. Just bind to `_` to silence the warning where you use it.

Reply:
Yes, ok.